src/v2.rs

```diff
@@ -169,12 +171,86 @@ impl EszipV2Modules {
   }
 }
 
+#[derive(Debug, Clone, Copy)]
+struct Options {
+  /// Hash Function used to checksum the module's source code.
+  ///
+  /// If the eszip does not include the option, it defaults to SHA256. If the parser does not know the hash function,
+  /// degrades to not checksuming
+  source_hash_function: Option<HashFunction>,
+
+  /// Size in Bytes of the hash function digest.
+  ///
+  /// Defaults to the known length of the configured hash function. Useful in order to ensure forwards compatibility,
+  /// otherwise the parser does not know how many bytes to read.
+  source_digest_size: Option<u8>,
+}
+
+impl Default for Options {
+  fn default() -> Self {
+    Self {
+      source_hash_function: Some(HashFunction::default()),
+      source_digest_size: Default::default(),
+    }
+  }
+}
+
+impl Options {
+  /// Get the size in Bytes of the source hashes
+  ///
+  /// If the eszip has an explicit digest size, returns that. Otherwise, returns
+  /// the default digest size of the [`Self::source_hash_function`]. If the eszip
+  /// does not have either, returns `None`.
+  fn source_digest_size(self) -> Option<u8> {
+    self
+      .source_digest_size
+      .or_else(|| Some(self.source_hash_function?.digest_size()))
+  }
+}
+
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq)]
+#[repr(u8)]
+pub enum HashFunction {
+  #[default]
+  Sha256 = 0,
+  Crc32 = 1,
+  XxHash = 2,
```

Comment:
Maybe we should choose one way to do hashing (on or off) then remove the options? (in the CLI it could assume the eszips have no hashing) Obviously backwards compatibility will require sha256 for old eszips.

Reply:
We might be able to chose the best hash function for today's context (use cases, hashing state of the art, etc), but we are unable to predict how the context might change in the future. IMO the explicit & flexible approach is worth it in this case.